src/CMakeLists.txt

```diff
@@ -18,6 +18,54 @@ mark_as_advanced(
 )
 add_subdirectory(crc32c EXCLUDE_FROM_ALL)
 
+set(LEVELDB_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+set(LEVELDB_BUILD_BENCHMARKS OFF CACHE BOOL "" FORCE)
+set(LEVELDB_INSTALL OFF CACHE BOOL "" FORCE)
+set(HAVE_CRC32C ON CACHE BOOL "" FORCE)
+set(HAVE_SNAPPY OFF CACHE BOOL "" FORCE)
+set(HAVE_TCMALLOC OFF CACHE BOOL "" FORCE)
+# Disabling TSA warnings because leveldb simultaneously
+# enables -Werror as a usage requirement.
+set(HAVE_CLANG_THREAD_SAFETY OFF CACHE BOOL "" FORCE)
+mark_as_advanced(
+  LEVELDB_BUILD_TESTS
+  LEVELDB_BUILD_BENCHMARKS
+  LEVELDB_INSTALL
+  HAVE_CRC32C
+  HAVE_SNAPPY
+  HAVE_TCMALLOC
+  HAVE_CLANG_THREAD_SAFETY
+)
+add_subdirectory(leveldb EXCLUDE_FROM_ALL)
+target_compile_definitions(leveldb
+  PRIVATE
+    FALLTHROUGH_INTENDED=[[fallthrough]]
+    $<$<CXX_COMPILER_ID:MSVC>:_CRT_NONSTDC_NO_WARNINGS>
+)
+target_compile_options(leveldb
+  PRIVATE
+    # RTTI has been forcibly disabled in leveldb's CMake configuration
```

Comment:
Exceptions are also disabled there, I assume that would cause sneaky problems as well?

This seems brittle to me.

Reply:
This code is [dropped](https://github.com/hebasto/bitcoin/pull/6#issuecomment-1435777703) now.